pkg/util/steps/refreshing.go

```diff
@@ -66,24 +66,23 @@ func (s authorizationRefreshingActionStep) run(ctx context.Context, log *logrus.
 		// We use the outer context, not the timeout context, as we do not want
 		// to time out the condition function itself, only stop retrying once
 		// timeoutCtx's timeout has fired.
-		err := s.step.run(ctx, log)
+		err := s.f(ctx)
 
-		// Don't refresh if we have timed out
-		if timeoutCtx.Err() == nil &&
-			(azureerrors.HasAuthorizationFailedError(err) ||
-				azureerrors.HasLinkedAuthorizationFailedError(err) ||
-				err == ErrWantRefresh) {
-			log.Print(err)
+		// If we haven't timed out and there is an error that is either an
+		// unauthorized client (AADSTS700016) or "AuthorizationFailed" (likely
+		// role propagation delay) then refresh and retry.
+		if timeoutCtx.Err() == nil && err != nil &&
+			(azureerrors.IsUnauthorizedClientError(err) ||
+				azureerrors.HasAuthorizationFailedError(err)) {
```

Comment:
nit: Unsure if this is in scope, but there's a drift here [between this repo and ARO-RP](https://github.com/Azure/ARO-RP/blob/master/pkg/util/steps/refreshing.go#L71-L78), which also tracks `azureerrors.IsInvalidSecretError(err)` - should we follow suit here?